src/checker.ml

```diff
@@ -126,43 +131,82 @@ let[@inline] ensure_valid_leaf_ptr (mem: mem) (leaf_ptr: leaf_ptr) : unit =
   | Some (Leaf _) -> ()
   | _ -> Ligo.failwith error_InvalidLeafPtr
 
-let[@inline] entrypoint_create_burrow (state, (burrow_no, delegate_opt): checker * (Ligo.nat * Ligo.key_hash option)) =
+(* TODO: if no tez is allowed to be given anywhere, it might be a better idea
```

Comment:
So, about this, I really want to try the sharing, but it will be quite a change for a few reasons:
* it affects _all_ entrypoints
* we have many tests that check per entrypoint that they fail if any tez is given which will have to be patched
* by moving code from entrypoint code (serialized) to checkerMain (non-serialized) gas costs might go up

So, I suggest that I open a separate PR after we've merged this one. What do you think @dorranh?